tests/sentry/api/test_event_search.py

```diff
@@ -525,40 +526,64 @@ def test_not_has_tag(self):
         ]
 
     def test_allowed_keys(self):
-        config = SearchConfig(allowed_keys=["good_key"])
+        config = SearchConfig.create_from(
+            default_config, allowed_keys=["good_key"], allow_free_text_email=True
+        )
```

Comment:
The tests might be easier to follow if we had new tests for the config where `allow_free_text_email=True`. Then within each test for allowed/blocked keys we could test search query text that contains no email-like string, and one that does contain an email-like string.